api/python/cellxgene_census/src/cellxgene_census/experimental/pp/_stats.py

```diff
@@ -0,0 +1,83 @@
+from __future__ import annotations
+
+import os
+from concurrent import futures
+from typing import Any, Generator, Tuple
+
+import numpy as np
+import numpy.typing as npt
+import pandas as pd
+import tiledbsoma as soma
+
+from ..util._eager_iter import _EagerIterator
+from ._online import MeanAccumulator, MeanVarianceAccumulator
+
+
+def mean_variance(
+    query: soma.ExperimentAxisQuery,
+    layer: str = "raw",
+    calculate_mean: bool = False,
+    calculate_variance: bool = False,
+) -> pd.DataFrame:
+    """
+    Calculate  mean and/or variance along the `obs` axis from query results. Calculations
+    are done in an accumulative chunked fashion.
+
+    Args:
+        query:
+            A SOMA query, specifying the obs/var selection over which mean and variance are calculated.
+
+        layer:
+            X layer used, e.g., `raw`
+
+        calculate_mean:
+            If `True it calculates mean, otherwise skips calculation
+
+        calculate_variance:
+            If `True it calculates variance, otherwise skips calculation
+
+    Returns:
+        Pandas DataFrame indexed by the obs `soma_joinid` and with columns
+       `mean` (if `calculate_mean = True`), and `variance` (if `calculate_variance = True`)
+
+    Lifecycle:
+        experimental
+    """
+
+    if calculate_mean is False and calculate_variance is False:
+        raise ValueError("At least one of `calculate_mean` or `calculate_variance` must be True")
+
+    n_batches = 1
+    n_samples = np.array([query.n_vars], dtype=np.int64)
+
+    def iterate() -> Generator[Tuple[npt.NDArray[np.int64], Any], None, None]:
+        max_workers = (os.cpu_count() or 4) + 2
+        with futures.ThreadPoolExecutor(max_workers=max_workers) as pool:
+            for arrow_tbl in _EagerIterator(query.X(layer).tables(), pool=pool):
+                obs_indexer = query.indexer
+                obs_dim = obs_indexer.by_obs(arrow_tbl["soma_dim_0"])
```

Comment:
performance note:  using the query indexer creates a Pandas.Index object. So you are creating two duplicate Index objects (one here, one on line 63), which are identical, and which are reasonably expensive if it is a large query.

As an alternative, create your own Index (i.e., line 63), and then use it to reindex.

Reply:
I see that the duplicate index is available as `query.indexer._obs_index` [which is private](https://github.com/single-cell-data/SOMA/blob/063aef6a32530c2f2730c73942490d3fc55916d3/python-spec/src/somacore/query/query.py#L525) - are we ok with using it even if private?

Reply:
I saw that but `by_obs` effectively forces a reindex, while here I need the "raw" `soma_joinid` index.